flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/ZooKeeperLeaderElectionConnectionHandlingTest.java

```diff
@@ -102,9 +102,10 @@ public void testConnectionSuspendedHandlingDuringInitialization() throws Excepti
 
 			closeTestServer();
 
-			CompletableFuture<String> secondAddress = queueLeaderElectionListener.next();
-			assertThat("No result is expected since there was no leader elected before stopping the server, yet.",
-				secondAddress, is(nullValue()));
+			// QueueLeaderElectionListener will be notified with an empty leader when ZK connection is suspended
+			final CompletableFuture<String> secondAddress = queueLeaderElectionListener.next(Duration.ofSeconds(3));
```

Comment:
If we are expecting an empty leader information, then a timeout is not required here.